modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.ClusterCachesInfo.CACHES_VIEW;
+import static org.apache.ignite.internal.processors.cache.ClusterCachesInfo.CACHE_GRPS_VIEW;
+import static org.apache.ignite.internal.processors.service.IgniteServiceProcessor.SVCS_VIEW;
+import static org.apache.ignite.internal.processors.task.GridTaskProcessor.TASKS_VIEW;
+
+/** */
+public class SystemViewSelfTest extends GridCommonAbstractTest {
+    /** */
+    private static CountDownLatch latch;
+
+    /** */
+    @Test
+    public void testCachesView() throws Exception {
+        try (IgniteEx g = startGrid()) {
+            Set<String> cacheNames = new HashSet<>(Arrays.asList("cache-1", "cache-2"));
+
+            for (String name : cacheNames)
+                g.createCache(name);
+
+            SystemView<CacheView> caches = g.context().metric().view(CACHES_VIEW);
+
+            assertEquals(g.context().cache().cacheDescriptors().size(), F.size(caches.iterator()));
+
+            for (CacheView row : caches)
+                cacheNames.remove(row.cacheName());
+
+            assertTrue(cacheNames.toString(), cacheNames.isEmpty());
+        }
+    }
+
+    /** */
+    @Test
+    public void testCacheGroupsView() throws Exception {
+        try(IgniteEx g = startGrid()) {
+            Set<String> grpNames = new HashSet<>(Arrays.asList("grp-1", "grp-2"));
+
+            for (String grpName : grpNames)
+                g.createCache(new CacheConfiguration<>("cache-" + grpName).setGroupName(grpName));
+
+            SystemView<CacheGroupView> grps = g.context().metric().view(CACHE_GRPS_VIEW);
+
+            assertEquals(g.context().cache().cacheGroupDescriptors().size(), F.size(grps.iterator()));
+
+            for (CacheGroupView row : grps)
+                grpNames.remove(row.cacheGroupName());
+
+            assertTrue(grpNames.toString(), grpNames.isEmpty());
+        }
+    }
+
+    /** */
+    @Test
+    public void testServices() throws Exception {
+        try(IgniteEx g = startGrid()) {
+            ServiceConfiguration srvcCfg = new ServiceConfiguration();
+
+            srvcCfg.setName("service");
+            srvcCfg.setMaxPerNodeCount(1);
+            srvcCfg.setService(new DummyService());
+
+            g.services().deploy(srvcCfg);
+
+            SystemView<ServiceView> srvs = g.context().metric().view(SVCS_VIEW);
+
+            assertEquals(g.context().service().serviceDescriptors().size(), F.size(srvs.iterator()));
+
+            ServiceView sview = srvs.iterator().next();
+
+            assertEquals(srvcCfg.getName(), sview.name());
+            assertEquals(srvcCfg.getMaxPerNodeCount(), sview.maxPerNodeCount());
+            assertEquals(DummyService.class, sview.serviceClass());
+        }
+    }
+
+    /** */
+    @Test
+    public void testComputeBroadcast() throws Exception {
```

Comment:
I think we should cover more cases for tasks (for example run tasks by name/by class/by closure, run tasks from another node), to see different values for different rows in one column of the list. The same for services.

Reply:
Fixed!!!